### PR TITLE
[#IC-51] Activation model

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### List of Changes

Create `ActivationModel` from `CosmosdbModelComposedVersioned`

#### Motivation and Context

To store services Activation for SpecialServices is introduced a new `ActivationModel`. This model handle versioning of documents identified by the tuple `fiscalCode` and `serviceId`.

#### How Has This Been Tested?

unit test
integration test

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/models/__tests__/activation.test.ts [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+  _ts: 1,
+  id: aFirstVersionId,
+  kind: "IRetrievedActivation",
+  version: 0 as NonNegativeInteger,
+  ...aRawActivation
+};
+
+describe("findLastVersionByModelId", () => {
+  it("should resolve to an existing activation", async () => {
+    const containerMock = ({
```
Comment (michaeldisaro) [inline src/models/__tests__/activation.test.ts:31]: I see this container mocked in every test. Maybe we could extract it and mock just the functions.

File: __integrations__/models/activation.test.ts [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+            })
+          );
+        }
+      ),
+      TE.toUnion
+    )();
+
+    // read latest version of the document
+    await pipe(
+      TE.taskEither.of<any, void>(void 0),
```
Comment (michaeldisaro) [inline __integrations__/models/activation.test.ts:73]: Why this task either of void?

**Review by michaeldisaro — APPROVED** [review]
Apart a minor change and a question code looks good and make sense.

File: src/models/__tests__/activation.test.ts (hunk 1 above)
Comment (BurnedMarshal) [reply src/models/__tests__/activation.test.ts]: Fixed in 31b83dd

**Review by michaeldisaro — APPROVED** [review]
LGTM

**Pull request merged**